### PR TITLE
215 feature/change log row color

### LINKED ISSUE #215: Change the color of the email log row if email failed to be sent

### PR DESCRIPTION
**Pull request by mariadanieldeepak**
Close #215

### CODE REVIEW

File: include/Core/UI/ListTable/LogListTable.php [hunk 1]
```diff
@@ -240,6 +243,25 @@ protected function column_cb( $item ) {
...
+	 * @since 2.4.0
+	 *
+	 * @access protected
+	 *
+	 * @param object $item
+	 *
+	 * @return string
+	 */
+	protected function column_sent_status( $item ) {
+		if ( 1 === absint( $item->result ) ) {
```
Comment (sudar) [inline include/Core/UI/ListTable/LogListTable.php]: What would happen for email logs that were captured before we implemented this field?

Will they now show as failed? Can we not show anything in that case?

Comment (mariadanieldeepak) [reply include/Core/UI/ListTable/LogListTable.php]: @sudar,

> What would happen for email logs that were captured before we implemented this field?

The resulting output will be the same as the failed emails.

Will they now show as failed? Can we not show anything in that case?

Okay, implemented.

image

**Comment by sudar** [comment]
@mariadanieldeepak 

Looks like Style CI tests are failing. Fix them and re-assign them for review.

File: include/Util/helper.php [hunk 2]
```diff
@@ -112,6 +122,144 @@ function checked_array( $values, $current ) {
...
+	if ( empty( $result ) ) {
+		return $log_row_classes[0];
+	}
+
+	$result = absint( $result );
+	if ( array_key_exists( $result, $log_row_classes ) ) {
+		return $log_row_classes[ $result ];
+	}
+
+	return $log_row_classes[0];
```
Comment (sudar) [inline include/Util/helper.php]: This entire function could be simplified to just

```suggestion
		if ( 1 === $result ) {
	    return 'el_email_sent_status--sent';
	}
	
	return 'el_email_sent_status--failed';
```

File: include/Util/helper.php [hunk 3]
```diff
@@ -112,6 +122,144 @@ function checked_array( $values, $current ) {
...
+}
+
+/**
+ * Gets the log row class by result code.
+ *
+ * @param int $result Mail sent status.
+ *
+ * @return string
+ */
+function get_log_row_class_by_result_code( $result ) {
```
[the remark this answers is not in the document]
Comment (mariadanieldeepak) [reply include/Util/helper.php]: @sudar,

It is not used and I missed to remove it.

**Pull request merged**